libs/core/src/main/java/org/opensearch/core/ParseField.java

```diff
@@ -43,7 +44,11 @@
 /**
  * Holds a field that can be found in a request while parsing and its different
  * variants, which may be deprecated.
+ *
+ * @opensearch.api
+ * 
  */
+@PublicApi(since = "1.0.0")
```

Comment:
This one is looks weird but totally legit: `Plugin` -> `ScriptService` -> `Script` -> : 
```
    /**
     * Declare a script field on an {@link ObjectParser}.
     * @param <T> Whatever type the {@linkplain ObjectParser} is parsing.
     * @param parser the parser itself
     * @param consumer the consumer for the script
     * @param parseField the field name
     */
    public static <T> void declareScript(AbstractObjectParser<T, ?> parser, BiConsumer<T, Script> consumer, ParseField parseField) {
        parser.declareField(consumer, (p, c) -> Script.parse(p), parseField, ValueType.OBJECT_OR_STRING);
    }
```